src/fips.c

```diff
@@ -1085,7 +1085,7 @@ ecdh_kat()
 	ica_adapter_handle_t ah;
 	const struct ecdh_kat_tv *tv;
 	unsigned char shared_secret[MAX_ECC_PRIV_SIZE];
-	ICA_EC_KEY *eckey_A, *eckey_B;
+	ICA_EC_KEY *eckey_A = NULL, *eckey_B = NULL;
```

Comment:
You must also set eckey_A and eckey_B to NULL after freeing it in lines 1134/1135.
Otherwise if it fails in the 2. loop iteration, eckey_B is still non-NULL, and gets freed twice.

Reply:
Hm, looks like the other solution was the better one...

Reply:
I see 2 options: initialize the pointers at the beginning of the for loop (instead of at the beginning of the function) or switch back to my first implementation. I personally prefer the latter one.

Reply:
Hm, sorry, yes, maybe your first solution was indeed better ...

Reply:
@jschmidb its up to you to decide.

Reply:
In this case it would not be necessary to initialize the keys to NULL, but I would do this anyway

Reply:
ok, I'll go back to the first implementation.